app/controllers/items_controller.rb

```diff
@@ -9,9 +9,14 @@ def edit
 
   def update
     item = Item.find(params[:id])
-    item.update(item_params)
-    flash[:success] = "#{item.name} has been successfully updated!"
-    redirect_to merchant_item_path
+    if params[:enable].nil? && params[:disable].nil?
```

Comment:
Have you thought about using a single parameter (:status)? - my thought being that with a single parameter that passes the exact value that will be updated you can pare down the conditional logic.